src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedCatchParameterShouldBeUnnamedCheckTest.java

```diff
@@ -0,0 +1,147 @@
+            "79:16: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "103:16: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputUnusedCatchParameterShouldBeUnnamed.java"),
+                expected);
+    }
+
+    @Test
+    public void testUnusedCatchParameterShouldBeUnnamedWithResourceAndFinally() throws Exception {
+        final String[] expected = {
+            "19:18: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "38:18: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "48:18: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "69:18: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath(
+                        "InputUnusedCatchParameterShouldBeUnnamedWithResourceAndFinally.java"),
+                expected);
+    }
+
+    @Test
+    public void testUnusedCatchParameterShouldBeUnnamedNested() throws Exception {
+        final String[] expected = {
+            "15:18: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "18:22: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "ex"),
+            "31:22: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "ex"),
+            "42:18: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "87:22: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "ex"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath(
+                        "InputUnusedCatchParameterShouldBeUnnamedNested.java"),
+                expected);
+    }
+
+    @Test
+    public void testUnusedCatchParameterShouldBeUnnamedInsideAnonClass() throws Exception {
+        final String[] expected = {
+            "14:18: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "39:28: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "50:18: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+            "57:28: " + getCheckMessage(MSG_UNUSED_CATCH_PARAMETER, "e"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath(
+                        "InputUnusedCatchParameterShouldBeUnnamedInsideAnonClass.java"),
+                expected);
+    }
+
+    @Test
+    public void testClearState() {
+        final UnusedCatchParameterShouldBeUnnamedCheck check =
+                new UnusedCatchParameterShouldBeUnnamedCheck();
+
+        final DetailAstImpl ast = new DetailAstImpl();
+        ast.setType(TokenTypes.LITERAL_CATCH);
+
+        final DetailAstImpl child = new DetailAstImpl();
+        child.setType(TokenTypes.PARAMETER_DEF);
+
+        final DetailAstImpl ident = new DetailAstImpl();
+        ident.setType(TokenTypes.IDENT);
+        ident.setText("e");
+        child.addChild(ident);
+
+        ast.addChild(child);
```

Comment:
Why can't we use this method of obtaining AST?
https://github.com/checkstyle/checkstyle/blob/1f85c8e9860b24235536b41ca9605c69945d3af4/src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheckTest.java#L332-L336

Reply:
done